### PR TITLE
chore: AR-28624129: Migrate to PNPm

### LINKED ISSUE #1: [Prettier] Migrate from NPM to PNPm
PNPm is our official package manager, so we must migrate all our projects from NPM to PNPm.

### Acceptance criteria:

In the [prettier-config](https://github.com/detra-lab/prettier-config) repo, do the following:

- Update the `README.md`
- Replace the `package-lock.json` with the `pnpm-lock.yaml`
- Update the GitHub Action

### Hints:

- Setup the GitHub Action with PNPm: https://youtu.be/eh89VE3Mk5g?t=120

### PR DESCRIPTION
**Pull request by giotramu**
Resolve #1

### CODE REVIEW

**Review by rosdb — DISMISSED** [review]
LGTM, check my comment before the merge.

File: package.json [hunk 1]
```diff
@@ -22,12 +22,21 @@
...
-    "prettier": "^2.8.0"
+    "@changesets/cli": "^2.26.1",
+    "prettier": "^2.8.8"
   },
   "peerDependencies": {
     "prettier": "^2.8.0"
   },
+  "peerDependenciesMeta": {
+    "prettier": {
+      "optional": false
```
[the remark this answers is not in the document]
Comment (giotramu) [reply package.json:37]: @rosdb, do you think it's unnecessary? 🤔

Maybe you [are right](https://docs.npmjs.com/cli/v7/configuring-npm/package-json#peerdependenciesmeta). I want to be more explicit, but perhaps it's just noise.

Thoughts?

[the remark this answers is not in the document]
Comment (rosdb) [reply package.json:37]: I'd remove it if it's the default behavior, we should avoid unnecessary information.

**Review by rosdb — APPROVED** [review]

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (giotramu) [reply package.json]: It should be fine now.

**Pull request merged**